### PR TITLE
overhaul app + create app env fe

### PR DESCRIPTION
**Pull request by jusrhee**
## What does this PR do?
old:
image

new:
image
image

### CODE REVIEW

File: dashboard/src/main/home/env-dashboard/EnvGroupArray.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export type KeyValueType = {
 type PropsType = {
   label?: string;
   values: KeyValueType[];
-  setValues: (x: KeyValueType[]) => void;
+  setValues?: (x: KeyValueType[]) => void;
```
[the remark this answers is not in the document]
Comment (ferozemohideen) [reply dashboard/src/main/home/env-dashboard/EnvGroupArray.tsx:24]: if you don't want to provide it in the prop, set a default in this component: `setValues = () => ({})`

File: dashboard/src/main/home/env-dashboard/EnvGroupArray.tsx [hunk 2]
```diff
@@ -293,7 +292,8 @@ const InputWrapper = styled.div`
   align-items: center;
   margin-bottom: 5px;
 `;
-const Input = styled.input<{ flex?: boolean }>`
+
+const Input = styled.input<{ flex?: boolean; override?: boolean }>`
```
Comment (ferozemohideen) [inline dashboard/src/main/home/env-dashboard/EnvGroupArray.tsx:296]: remove override

File: dashboard/src/main/home/app-dashboard/validate-apply/app-settings/EnvGroupRow.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    ).map(([key, value]) => ({
+      key,
+      value,
+      hidden: true,
+      locked: true,
+      deleted: false,
+    }));
+    const variables = [...normalVariables, ...secretVariables];
+    setEnvVariables(variables);
+  }, [envGroup]);
```
[the remark this answers is not in the document]
Comment (jusrhee) [reply dashboard/src/main/home/app-dashboard/validate-apply/app-settings/EnvGroupRow.tsx:48]: memoized but looks like those values are actually used by the old envgrouparray so will keep for now

File: dashboard/src/main/home/app-dashboard/validate-apply/app-settings/EnvVarRow.tsx [hunk 4]
```diff
@@ -22,83 +25,60 @@ const EnvVarRow: React.FC<Props> = ({
...
   const { control: appControl, watch } = useFormContext<PorterAppFormData>();
   const hidden = watch(`app.env.${index}.hidden`);
-  const keys = watch(`app.env.${index}.key`);
-
-  const validKey = (key: string): boolean => /^[A-Za-z]/.test(key);
 
   return (
     <InputWrapper>
       {entry.locked ? (
-        <Tooltip
```
[the remark this answers is not in the document]
Comment (jusrhee) [reply dashboard/src/main/home/app-dashboard/validate-apply/app-settings/EnvVarRow.tsx:32]: part of it rendered offscreen / looked really janky

**Pull request merged**